### PR TITLE
check if merge key exists when splitting yaml

### PR DESCRIPTION
**Pull request by TianqiZhang**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs [hunk 1]
```diff
@@ -102,8 +102,15 @@ public SchemaFragmentsIterator(ISchemaFragmentsHandler handler)
...
-                            var opath = string.Format("{0}[{1}=\"{2}\"]", parentOPath, mergeKey, mapNode.Children[mergeKey].ToString());
-                            TraverseCore(item, fragments, schema.Items, opath, uid);
+                            if (mapNode.Children.ContainsKey(mergeKey))
+                            {
+                                var opath = string.Format("{0}[{1}=\"{2}\"]", parentOPath, mergeKey, mapNode.Children[mergeKey].ToString());
+                                TraverseCore(item, fragments, schema.Items, opath, uid);
+                            }
+                            else
+                            {
+                                Logger.LogError($"Cannot find merge key {mergeKey} in {mapNode.ToString()}");
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: Do you need return here to terminate the method?

Comment (herohua) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: >.ToString() [](start = 94, length = 11)

no need

Comment (herohua) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: >LogError [](start = 39, length = 8)

@superyyrrzz@gmail.com 
When there is error, will the caller of SchemaFragmentsIterator terminate?

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: No, only an error is logged.

**Comment by herohua** [comment]
add unit test

File: src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs (hunk 1 above)
Comment (herohua) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: is it expected?

---
In reply to: [200884075](https://github.com/dotnet/docfx/pull/3019#discussion_r200884075) [](ancestors = 200884075)

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs]: Seems SchemaFragmentsIterator should throw, while the call should catch and handle it. I'm not sure whether exception is needed here. Will talk to Tianqi about this.

**Comment by TianqiZhang** [comment]
thanks guys for reviewing this, as the PR's purpose is just to expose more information when the certain error occurs, UT will be added later.

**Pull request merged**